File: Source/ses_io/src/wrappers/fortran/wrap_ses_read_pairs.f90


#undef DEBUG_WRAP

SES_ERROR_FLAG function ses_read_pairs_f90(the_handle, buf1, buf2, dim1)

   implicit none

   SES_FILE_HANDLE, INTENT(IN) :: the_handle
   SES_WORD_REFERENCE :: buf1
   SES_WORD_REFERENCE :: buf2
   SES_NUMBER, intent(IN) :: dim1   

   SES_ERROR_FLAG :: return_value, ses_read_pairs

#ifdef DEBUG_WRAP
   print *, "ses_read_pairs_f90:  the_handle is ", the_handle
   print *, "ses_read_pairs_f90:  dim1 is ", dim1
#endif

   return_value = ses_read_pairs(the_handle, buf1, buf2, dim1)

#ifdef DEBUG_WRAP

   print *, "ses_read_pairs_f90:  buf1(1) is ", buf1(1)
   print *, "ses_read_pairs_f90:  buf2(1) is ", buf2(1)
#endif

   ses_read_pairs_f90 = return_value

 
end function ses_read_pairs_f90

File: Source/ses_io/src/wrappers/fortran/wrap_ses_read_pairs.c


#include "ses_defines.h"
#include "ses_globals.h"
#include "ses_internals.h"
#include "ses_externs.h"

#include "stdio.h"
#include <stdlib.h>

#include <stdio.h>

#undef DEBUG_WRAP

#ifdef LC_UNDER
ses_error_flag ses_read_pairs_(ses_file_handle* pt_handle, ses_word_reference buf1, ses_word_reference buf2, ses_number* pt_dim1) {
#else

#ifdef UC_UNDER
ses_error_flag SES_READ_PAIRS_(ses_file_handle* pt_handle, ses_word_reference buf1, ses_word_reference buf2, ses_number* pt_dim1) {
#else

#ifdef LC_NOUNDER
ses_error_flag ses_READ_PAIRS(ses_file_handle* pt_handle, ses_word_reference buf1, ses_word_reference buf2, ses_number* pt_dim1) {
#else

#ifdef UC_NOUNDER
ses_error_flag SES_READ_PAIRS(ses_file_handle* pt_handle, ses_word_reference buf1, ses_word_reference buf2, ses_number* pt_dim1)  {
#endif

#endif
#endif
#endif

  ses_error_flag return_value = SES_NO_ERROR;

  ses_file_handle the_handle = *pt_handle;
  ses_number dim1 = *pt_dim1;
#ifdef DEBUG_WRAP
  printf("wrap_ses_read_pairs.c:  the_handle is %d\n", the_handle);
  printf("wrap_ses_read_pairs.c:  dim1 is %d\n", dim1);
#endif

  return_value = ses_read_pairs(the_handle, buf1, buf2, dim1);
#ifdef DEBUG_WRAP 
  printf("wrap_ses_read_pairs.c:  return_value is %d\n", return_value);
#endif
  return return_value;

}

 
